src/briefcase/integrations/android_sdk.py

```diff
@@ -212,61 +248,305 @@ def verify_emulator(self):
+            running. If ``device`` is null, a new emulator should be created.
+        """
+        # Get the list of attached devices (includes running emulators)
+        running_devices = self.devices()
+
+        # Choices is an ordered list of options that can be shown to the user.
+        # Each device should appear only once, and be keyed by AVD only if
+        # a device ID isn't available.
+        choices = []
+        # Device choices is the full lookup list. Devices can be looked up
+        # by any valid key - ID *or* AVD.
+        device_choices = {}
+
+        # Iterate over all the running devices.
+        # If the device has an AVD, use ADB to get the emulator AVD name.
+        # If it is a physical device, use the device name.
+        # Keep a log of all running AVDs
+        running_avds = {}
+        for d, details in sorted(running_devices.items(), key=lambda d: d[1]["name"]):
+            name = details["name"]
+            avd = self.adb(d).avd_name()
+            if avd:
+                # It's a running emulator
+                running_avds[avd] = d
+                full_name = "@{avd} (running {name} emulator)".format(
+                    avd=avd, name=name,
+                )
+                choices.append((d, full_name))
+
+                # Save the AVD as a device detail.
+                details["avd"] = avd
+
+                # Device can be looked up by device ID or AVD
+                device_choices[d] = full_name
+                device_choices["@" + avd] = full_name
+            else:
+                # It's a physical device (might be disabled)
+                choices.append((d, name))
+                device_choices[d] = name
+
+        # Add any non-running emulator AVDs to the list of candidate devices
+        for avd in self.emulators():
+            if avd not in running_avds:
+                name = "@{avd} (emulator)".format(avd=avd)
+                choices.append(("@" + avd, name))
+                device_choices["@" + avd] = name
+
+        # If a device or AVD has been provided, check it against the available
+        # device list.
+        if device_or_avd:
+            try:
+                name = device_choices[device_or_avd]
+
+                if device_or_avd.startswith("@"):
+                    # specifier is an AVD
+                    try:
+                        avd = device_or_avd[1:]
+                        device = running_avds[avd]
+                    except KeyError:
+                        # device_or_avd isn't in the list of running avds;
+                        # it must be a non-running emulator.
+                        return None, name, avd
+                else:
+                    # Specifier is a direct device ID
+                    avd = None
+                    device = device_or_avd
+
+                details = running_devices[device]
+                avd = details.get("avd")
+                if details["authorized"]:
```

Comment:
Since `adb` only ever prints `unauthorized`, I'd prefer that `details` store `unauthorized=True` rather than `authorized=False`, to avoid making up our own nomenclature. That's just personal preference.

Reply:
Seems a bit bikeshed to me. It works both ways, and "not unauthorized" would be a double negative...